Build failure: mysql-workbench-8.0.38

### Steps To Reproduce

Steps to reproduce the behavior:
1. build mysql-workbench-8.0.38

### Build log

```
 > FAILED: tools/genwrap/genwrap
       > : && /nix/store/62zpnw69ylcfhcpy1di8152zlzmbls91-gcc-wrapper-13.3.0/bin/g++ -O3 -DNDEBUG  tools/genwrap/CMakeFiles/genwrap.dir/genwrap.cpp.o -o tools/genwrap/genwrap  -Wl,-rpath,/build/mysql-workbench-community-8.0.38-src/build/backend/wbpublic:/build/mysql-workbench-community-8.0.38-src/build/library/mtemplate:/build/mysql-workbench-community-8.0.38-src/build/library/forms:/build/mysql-workbench-community-8.0.38-src/build/ext/scintilla/gtk:/build/mysql-workbench-community-8.0.38-src/build/frontend/linux/linux_utilities:/build/mysql-workbench-community-8.0.38-src/build/library/grt/src:/build/mysql-workbench-community-8.0.38-src/build/library/mysql.canvas/src:/build/mysql-workbench-community-8.0.38-src/build/library/base:/build/mysql-workbench-community-8.0.38-src/build/library/cdbc:/build/mysql-workbench-community-8.0.38-src/build/library/parsers  backend/wbpublic/libwbpublic.so.8.0.38  library/mtemplate/libmtemplate.so.8.0.38  /nix/store/j0gcrmxywvlkpvjm4vxv468w5bjdpwyz-libSM-1.2.4/lib/libSM.so  /nix/store/g1i1aj007vn5p39bk1km6sfl1g20vsyr-libICE-1.1.1/lib/libICE.so  /nix/store/94sinl3kmf4byyjgfpy4mmm9crwj8imc-libX11-1.8.9/lib/libX11.so  /nix/store/053wlpj13qsrvq01fmky8gjj4amz2wbj-libXext-1.3.6/lib/libXext.so  -L/nix/store/249lkgncykrn2q61b3xawnf3jfifjhr1-mysql-8.0.38/lib -lmysqlclient -lzstd -lssl -lcrypto -lresolv -lm  -L/nix/store/7sxl9ysz2rkm2ln1r505ah2jknkkfrps-vsqlite-0.3.13/lib -lvsqlitepp  library/grt/src/libgrt.so.8.0.38  /nix/store/amlkmz4vr2j0khjnwfdx25glbd0mcxjj-libxml2-2.13.2/lib/libxml2.so  -luuid  library/base/libwbbase.so.8.0.38  -L/nix/store/wc9bysi0qgb0rw3lfslj72czxqg5x31j-antlr-runtime-cpp-4.12.0/lib -lantlr4-runtime  -Wl,-rpath-link,/build/mysql-workbench-community-8.0.38-src/build/library/forms:/build/mysql-workbench-community-8.0.38-src/build/ext/scintilla/gtk:/build/mysql-workbench-community-8.0.38-src/build/frontend/linux/linux_utilities:/build/mysql-workbench-community-8.0.38-src/build/library/mysql.canvas/src:/build/mysql-workbench-community-8.0.38-src/build/library/cdbc:/build/mysql-workbench-community-8.0.38-src/build/library/parsers && :
       > /nix/store/qsx2xqqm0lp6d8hi86r4y0rz5v9m62wn-binutils-2.42/bin/ld: /nix/store/gsa6jmsac2gqyfazlfncbj251q936wqi-libspatialite-5.1.0/lib/libspatialite.so.8: undefined reference to `xmlNanoHTTPCleanup@LIBXML2_2.4.30'
       > collect2: error: ld returned 1 exit status
       > [763/788] Linking CXX executable plugins/migration/wbcopytables-bin
       > FAILED: plugins/migration/wbcopytables-bin
       > : && /nix/store/62zpnw69ylcfhcpy1di8152zlzmbls91-gcc-wrapper-13.3.0/bin/g++ -O3 -DNDEBUG  plugins/migration/CMakeFiles/wbcopytables-bin.dir/copytable/copytable.cpp.o plugins/migration/CMakeFiles/wbcopytables-bin.dir/copytable/python_copy_data_source.cpp.o plugins/migration/CMakeFiles/wbcopytables-bin.dir/copytable/main.cpp.o plugins/migration/CMakeFiles/wbcopytables-bin.dir/copytable/converter.cpp.o -o plugins/migration/wbcopytables-bin  -Wl,-rpath,/build/mysql-workbench-community-8.0.38-src/build/backend/wbprivate:/build/mysql-workbench-community-8.0.38-src/build/backend/wbpublic:/build/mysql-workbench-community-8.0.38-src/build/library/ssh:/build/mysql-workbench-community-8.0.38-src/build/library/cdbc:/build/mysql-workbench-community-8.0.38-src/build/library/parsers:/build/mysql-workbench-community-8.0.38-src/build/library/forms:/build/mysql-workbench-community-8.0.38-src/build/ext/scintilla/gtk:/build/mysql-workbench-community-8.0.38-src/build/frontend/linux/linux_utilities:/build/mysql-workbench-community-8.0.38-src/build/library/grt/src:/build/mysql-workbench-community-8.0.38-src/build/library/mysql.canvas/src:/build/mysql-workbench-community-8.0.38-src/build/library/mtemplate:/build/mysql-workbench-community-8.0.38-src/build/library/base:  backend/wbprivate/libwbprivate.so.8.0.38  backend/wbpublic/libwbpublic.so.8.0.38  -lgmodule-2.0  -lglib-2.0  -L/nix/store/249lkgncykrn2q61b3xawnf3jfifjhr1-mysql-8.0.38/lib -lmysqlclient -lzstd -lssl -lcrypto -lresolv -lm  -L/nix/store/95ajbz6dzb20wfjgr0x3bqg762azzb7z-libiodbc-3.52.16/lib  -liodbc -liodbcinst  /nix/store/l014xp1qxdl6gim3zc0jv3mpxhbp346s-python3-3.12.4/lib/libpython3.12.so  /nix/store/amlkmz4vr2j0khjnwfdx25glbd0mcxjj-libxml2-2.13.2/lib/libxml2.so  -lzip  library/ssh/libwbssh.so.8.0.38  /nix/store/vr7vcfbwbb48h9cnd1m4w1z1yyxgyd2p-libssh-0.10.6/lib/libssh.so.4.9.6  /nix/store/8cjxwskigv1smh9hvd82hbr65p4fgjcl-gdal-3.9.1/lib/libgdal.so  -L/nix/store/7sxl9ysz2rkm2ln1r505ah2jknkkfrps-vsqlite-0.3.13/lib -lvsqlitepp  -L/nix/store/wc9bysi0qgb0rw3lfslj72czxqg5x31j-antlr-runtime-cpp-4.12.0/lib -lantlr4-runtime  library/base/libwbbase.so.8.0.38  -lglib-2.0  /nix/store/l014xp1qxdl6gim3zc0jv3mpxhbp346s-python3-3.12.4/lib/libpython3.12.so  /nix/store/amlkmz4vr2j0khjnwfdx25glbd0mcxjj-libxml2-2.13.2/lib/libxml2.so  -lzip  -Wl,-rpath-link,/build/mysql-workbench-community-8.0.38-src/build/library/cdbc:/build/mysql-workbench-community-8.0.38-src/build/library/parsers:/build/mysql-workbench-community-8.0.38-src/build/library/forms:/build/mysql-workbench-community-8.0.38-src/build/ext/scintilla/gtk:/build/mysql-workbench-community-8.0.38-src/build/frontend/linux/linux_utilities:/build/mysql-workbench-community-8.0.38-src/build/library/grt/src:/build/mysql-workbench-community-8.0.38-src/build/library/mysql.canvas/src:/build/mysql-workbench-community-8.0.38-src/build/library/mtemplate && :
       > /nix/store/qsx2xqqm0lp6d8hi86r4y0rz5v9m62wn-binutils-2.42/bin/ld: /nix/store/gsa6jmsac2gqyfazlfncbj251q936wqi-libspatialite-5.1.0/lib/libspatialite.so.8: undefined reference to `xmlNanoHTTPCleanup@LIBXML2_2.4.30'
       > collect2: error: ld returned 1 exit status
```

### Additional context

Might be related to https://github.com/NixOS/nixpkgs/pull/330337/files or https://github.com/NixOS/nixpkgs/pull/326637/files

### Notify maintainers

@TomaSajt 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.10.1, NixOS, 24.11 (Vicuna), 24.11.20240727.b73c222`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.23.3`
 - channels(root): `""`
 - nixpkgs: `/nix/store/qxf6anli54ij0q1sdlnlgx9hyl658a4v-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/misc/mysql-workbench/default.nix
+++ b/pkgs/applications/misc/mysql-workbench/default.nix
@@ -105,7 +105,7 @@ stdenv.mkDerivation (finalAttrs: {
     antlr4_12.runtime.cpp
     python3
     mysql
-    libxml2
+    (libxml2.override { enableHttp = true; })
     libmysqlconnectorcpp
     vsqlite
     gdal

